### PR TITLE
Allow a default retry backoff to be configured

### PR DESCRIPTION
**Pull request by odlp**
Hello,

Thanks for Job Iteration – it's awesome.

This PR proposes introducing a default retry backoff. This can be used to introduce a delay before an interrupted job is resumed. The delay still defaults to `nil` (immediate retry) to maintain existing behaviour, and the default will be overridden by a throttle-enumerator if applicable.

## Motivation

We've been using Job Iteration in production for a while and have noticed a weird interaction with Sidekiq Pro's [super fetch](https://github.com/sidekiq/sidekiq/wiki/Reliability#using-super_fetch) mode. There appears to be a race condition between: 

- The job being re-enqueued. This involves the super fetcher moving the job from a workers private queue back to a general queue.
- The job being picked up again by the same worker.

This appears to be responsible for the following warning from Sidekiq (and is preceded by a job interrupting itself):

```
ERROR: Unable to remove job from private queue queue:sq|ip-10-132-216-198.eu-west-1.compute.internal:94:c14bb11fc23a|lowest: 0
```

The main side-effect of this (apart from the warning) is that iterations can end up being repeated, presumably because the worker is using a stale version of the job payload.

## Reproducing the issue

I've been trying to work on an isolated reproduction script, but it doesn't always exhibit the issue. 

With a job like:

```ruby
# frozen_string_literal: true

class LongJob < ApplicationJob
  include JobIteration::Iteration

  def build_enumerator(cursor:)
    enumerator_builder.build_times_enumerator(
      1_000_000,
      cursor: cursor,
    )
  end

  def each_iteration(value)
    Sidekiq.logger.info("Processing #{value}")
    sleep(1)
  end
end
```

And Job Iteration configured to interrupt jobs every 20s (for issue-repro purposes only):

```ruby
JobIteration.max_job_runtime = 20
```

When Sidekiq `super_fetch!` is enabled it's possible to intermittently replicate the issue. Here's some log output when I observed the issue locally:

<details>
<summary>Expand log excerpt</summary>
<pre>
2023-01-25T15:24:22.085Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: {:@message=>"Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:24:20Z", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:20Z"}}
2023-01-25T15:24:22.576Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 0
2023-01-25T15:24:23.578Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 1
2023-01-25T15:24:24.580Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 2
2023-01-25T15:24:25.582Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 3
2023-01-25T15:24:26.585Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 4
2023-01-25T15:24:27.590Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 5
2023-01-25T15:24:28.593Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 6
2023-01-25T15:24:29.599Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 7
2023-01-25T15:24:30.602Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 8
2023-01-25T15:24:31.607Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 9
2023-01-25T15:24:32.610Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 10
2023-01-25T15:24:33.614Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 11
2023-01-25T15:24:34.615Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 12
2023-01-25T15:24:35.617Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 13
2023-01-25T15:24:36.619Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 14
2023-01-25T15:24:37.621Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 15
2023-01-25T15:24:38.622Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 16
2023-01-25T15:24:39.624Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 17
2023-01-25T15:24:40.626Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 18
2023-01-25T15:24:41.628Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 19
[ActiveJob] [LongJob] [180e638d-…] [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=19
2023-01-25T15:24:42.633Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=19
[ActiveJob] [LongJob] [180e638d-…] Enqueued LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) to Sidekiq(default)
[ActiveJob] [LongJob] [180e638d-…] Retrying LongJob in 0 seconds.
2023-01-25T15:24:42.741Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: {:@message=>"Retrying LongJob in 0 seconds.", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:20Z"}}
[ActiveJob] [LongJob] [180e638d-…] Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20665.9ms
2023-01-25T15:24:42.751Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: {:@message=>"Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20665.9ms", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:20Z", :duration=>"20665.9ms"}, :@primary=>true}
2023-01-25T15:24:42.753Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 elapsed=21.121 INFO: done
2023-01-25T15:24:42.754Z pid=73547 tid=pnb ERROR: Unable to remove job from private queue queue:sq|<hostname>.local:73547:8987852ea841|default: 0
2023-01-25T15:24:42.756Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 INFO: start
[ActiveJob] [LongJob] [180e638d-…] Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:24:20Z
2023-01-25T15:24:43.043Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: {:@message=>"Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:24:20Z", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:20Z"}}
2023-01-25T15:24:43.409Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 0
2023-01-25T15:24:44.411Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 1
2023-01-25T15:24:45.414Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 2
2023-01-25T15:24:46.417Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 3
2023-01-25T15:24:47.422Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 4
2023-01-25T15:24:48.426Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 5
2023-01-25T15:24:49.427Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 6
2023-01-25T15:24:50.428Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 7
2023-01-25T15:24:51.432Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 8
2023-01-25T15:24:52.436Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 9
2023-01-25T15:24:53.439Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 10
2023-01-25T15:24:54.444Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 11
2023-01-25T15:24:55.448Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 12
2023-01-25T15:24:56.452Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 13
2023-01-25T15:24:57.458Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 14
2023-01-25T15:24:58.463Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 15
2023-01-25T15:24:59.467Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 16
2023-01-25T15:25:00.469Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 17
2023-01-25T15:25:01.471Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 18
2023-01-25T15:25:02.473Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: Processing 19
[ActiveJob] [LongJob] [180e638d-…] [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=19
2023-01-25T15:25:03.477Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=19
[ActiveJob] [LongJob] [180e638d-…] Enqueued LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) to Sidekiq(default)
[ActiveJob] [LongJob] [180e638d-…] Retrying LongJob in 0 seconds.
[ActiveJob] [LongJob] [180e638d-…] Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20471.63ms
2023-01-25T15:25:03.515Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 INFO: {:@message=>"Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20471.63ms", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:20Z", :duration=>"20471.63ms"}, :@primary=>true}
2023-01-25T15:25:03.516Z pid=73547 tid=pnb class=LongJob jid=434318e41f34415f7326c007 attempts=1 elapsed=20.759 INFO: done
2023-01-25T15:25:03.519Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 INFO: start
[ActiveJob] [LongJob] [180e638d-…] Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:24:42Z
2023-01-25T15:25:03.834Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:24:42Z", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:42Z"}}
2023-01-25T15:25:04.110Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 20
2023-01-25T15:25:05.113Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 21
2023-01-25T15:25:06.117Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 22
2023-01-25T15:25:07.123Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 23
2023-01-25T15:25:08.123Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 24
2023-01-25T15:25:09.124Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 25
2023-01-25T15:25:10.125Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 26
2023-01-25T15:25:11.128Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 27
2023-01-25T15:25:12.133Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 28
2023-01-25T15:25:13.136Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 29
2023-01-25T15:25:14.140Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 30
2023-01-25T15:25:15.143Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 31
2023-01-25T15:25:16.145Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 32
2023-01-25T15:25:17.147Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 33
2023-01-25T15:25:18.148Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 34
2023-01-25T15:25:19.149Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 35
2023-01-25T15:25:20.153Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 36
2023-01-25T15:25:21.156Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 37
2023-01-25T15:25:22.161Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 38
2023-01-25T15:25:23.162Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 39
[ActiveJob] [LongJob] [180e638d-…] [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=39
2023-01-25T15:25:24.162Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=39
[ActiveJob] [LongJob] [180e638d-…] Enqueued LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) to Sidekiq(default)
[ActiveJob] [LongJob] [180e638d-…] Retrying LongJob in 0 seconds.
2023-01-25T15:25:24.195Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Retrying LongJob in 0 seconds.", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:42Z"}}
[ActiveJob] [LongJob] [180e638d-…] Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20366.94ms
2023-01-25T15:25:24.201Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20366.94ms", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:24:42Z", :duration=>"20366.94ms"}, :@primary=>true}
2023-01-25T15:25:24.201Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 elapsed=20.682 INFO: done
2023-01-25T15:25:24.206Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 INFO: start
[ActiveJob] [LongJob] [180e638d-…] Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:25:03Z
2023-01-25T15:25:24.504Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:25:03Z", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:25:03Z"}}
2023-01-25T15:25:24.776Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 20
2023-01-25T15:25:25.778Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 21
2023-01-25T15:25:26.781Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 22
2023-01-25T15:25:27.784Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 23
2023-01-25T15:25:28.786Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 24
2023-01-25T15:25:29.787Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 25
2023-01-25T15:25:30.792Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 26
2023-01-25T15:25:31.795Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 27
2023-01-25T15:25:32.798Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 28
2023-01-25T15:25:33.803Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 29
2023-01-25T15:25:34.808Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 30
2023-01-25T15:25:35.812Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 31
2023-01-25T15:25:36.815Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 32
2023-01-25T15:25:37.819Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 33
2023-01-25T15:25:38.825Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 34
2023-01-25T15:25:39.829Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 35
2023-01-25T15:25:40.832Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 36
2023-01-25T15:25:41.836Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 37
2023-01-25T15:25:42.839Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 38
2023-01-25T15:25:43.844Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 39
[ActiveJob] [LongJob] [180e638d-…] [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=39
2023-01-25T15:25:44.848Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=39
[ActiveJob] [LongJob] [180e638d-…] Enqueued LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) to Sidekiq(default)
2023-01-25T15:25:44.880Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Enqueued LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) to Sidekiq(default)", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:25:03Z"}}
[ActiveJob] [LongJob] [180e638d-…] Retrying LongJob in 0 seconds.
2023-01-25T15:25:44.881Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Retrying LongJob in 0 seconds.", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:25:03Z"}}
[ActiveJob] [LongJob] [180e638d-…] Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20383.9ms
2023-01-25T15:25:44.888Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Performed LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) in 20383.9ms", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:25:03Z", :duration=>"20383.9ms"}, :@primary=>true}
2023-01-25T15:25:44.888Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 elapsed=20.683 INFO: done
2023-01-25T15:25:44.893Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 INFO: start
[ActiveJob] [LongJob] [180e638d-…] Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:25:24Z
2023-01-25T15:25:45.175Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: {:@message=>"Performing LongJob (Job ID: 180e638d-fa43-4328-9a87-2f2293428d31) from Sidekiq(default) enqueued at 2023-01-25T15:25:24Z", :@fields=>{:job_id=>"180e638d-fa43-4328-9a87-2f2293428d31", :queue_name=>"default", :attempts=>1, :enqueued_at=>"2023-01-25T15:25:24Z"}}
2023-01-25T15:25:45.445Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 40
2023-01-25T15:25:46.449Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 41
2023-01-25T15:25:47.451Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 42
2023-01-25T15:25:48.454Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 43
2023-01-25T15:25:49.459Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 44
2023-01-25T15:25:50.464Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 45
2023-01-25T15:25:51.468Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 46
2023-01-25T15:25:52.471Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 47
2023-01-25T15:25:53.475Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 48
2023-01-25T15:25:54.480Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 49
2023-01-25T15:25:55.481Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 50
2023-01-25T15:25:56.482Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 51
2023-01-25T15:25:57.484Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 52
2023-01-25T15:25:58.485Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 53
2023-01-25T15:25:59.486Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 54
2023-01-25T15:26:00.489Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 55
2023-01-25T15:26:01.490Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 56
2023-01-25T15:26:02.495Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 57
2023-01-25T15:26:03.500Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 58
2023-01-25T15:26:04.503Z pid=73547 tid=pnb class=LongJob jid=180e638d-fa43-4328-9a87-2f2293428d31 attempts=1 INFO: Processing 59
[ActiveJob] [LongJob] [180e638d-…] [JobIteration::Iteration] Interrupting and re-enqueueing the job cursor_position=59
</pre>
</details>

In summary: 

- 0-19 (first pass)
- 0-19 (repeat 😢)
- 20-39 (first pass)
- 20-39 (repeat 😢)
- 40-50 (first pass)

## Mitigation

Introducing a small delay before a job is retried appears to mitigate the issue (hence this PR!), although I still need to figure out the underlying cause (which could well be in Sidekiq / ActiveJob, not Job Iteration).

### CODE REVIEW

**Comment by odlp** [comment]
I've now signed the CLA, but don't have permissions to re-run the check.
